rust/tools/authorization-logic/src/souffle/lowering_ast_datalog.rs

```diff
@@ -0,0 +1,305 @@
+use crate::ast::*;
+use crate::souffle::datalog_ir::*;
+
+// Asside from queries, this translation is quite similar to the one given in 
```

Comment:
As with other cases, should probalby be module level doc comments (https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#commenting-contained-items)

Reply:
Suggestion applied broadly.